src/transformState.js

```diff
@@ -5,7 +5,27 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    for (const key in action) {
+      if (action[key] === 'addProperties') {
```

Comment:
```suggestion
    for (const key in action) {
      if (action[key] === 'addProperties') {
```
redundant for in loop. destructure action and check type
const { type, extraData , keysToRemove } = action;

Reply:
also instead if consider to use switch/case 
```
  switch(type) {
    //do something
  }